### DISCUSSION TITLE
How to build a custom temporal graph dataset ? Combining GCN with LSTMcells

### DISCUSSION THREAD

**Discussion by pigu163** [pyg-team/pytorch_geometric#6791] [Q&A]
Hello,

I am new at PyG, currenly I am using GATconv to extract features then put them into each LSTMcell, and the graph should be dynamic(edges might be changed over time, features also) 

My question is, how can I customize my dataset by using PyG? Obviously using origin torch dataset and data_loader would be easy to form a LSTM usage dataset but seems temporal graph is different. Beside, Batch mechanisum looks different between PyG and torch, could they combine in some ways?

**Comment by rusty1s** [comment]
I think what people are usually doing is run your GCN in parallel on each timestamp (using PyG mini-batching), and then convert the node embeddings to shape `[num_timestamps, num_nodes, num_features]` via `utils.to_dense_batch`. Then, you can use an LSTM to process this sequence of node embeddings.

**Reply by pigu163** [reply]
Hi, Thank you for your reply. For instance, if I have 50 length timestamp, do I need 50 GCNs ?

**Reply by rusty1s** [reply]
I don't think so. IMO, the GCN can be shared across timestamps.

**Reply by pigu163** [reply]
Thank you for your hint! That's useful!
